library/pk_wrap.c

```diff
@@ -571,15 +571,15 @@ static int ecdsa_verify_wrap( void *ctx, mbedtls_md_type_t md_alg,
     if( key_len <= 0 )
         return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
 
-    if( ( ret = mbedtls_psa_get_free_key_slot( &key_slot ) ) != PSA_SUCCESS )
-        return( mbedtls_psa_err_translate_pk( ret ) );
-
     psa_md = mbedtls_psa_translate_md( md_alg );
     if( psa_md == 0 )
         return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
     psa_sig_md = PSA_ALG_ECDSA( psa_md );
     psa_type = PSA_KEY_TYPE_ECC_PUBLIC_KEY( curve );
 
+    if( ( ret = psa_allocate_key( psa_type, key_len * 8, &key_slot ) ) != PSA_SUCCESS )
```

Comment:
We're passing the byte size of the exported key here, which is strictly larger than the bit size of the actual key. Reading the documentation of `psa_allocate_key()`, such overapproximation is fine, but I think it would be sufficient to pass the bit size of the ECC curve here. @gilles-peskine-arm Is that correct?

Reply:
TL,DR: pass the bit size of the ECC curve for correctness. Pass whatever you like to work with Mbed Crypto in the short term.

At the moment the specification of `psa_allocate_key` is vague and I don't have a clear picture of what it should be. The reason `psa_allocate_key` takes arguments is that some implementations that want to know how much memory to allocate, for example `psa_create_key` in a secure element. It not relevant to Mbed Crypto at the moment since it will call `malloc` anyway.

In principle, an implementation of the PSA crypto API should be allowed to say “a 2048-bit ECC key? Nope, I can't do that”. It's a resource usage vs convenience compromise that I think the specification should leave up to implementations.

In the case of ECC keys, I'd like to allow applications to just say “I'm going to use this curve” and not have to worry about a size. On the other hand, for an RSA key, the size is important here to allow implementation that don't use malloc. This is still an open design topic (private discussion at https://github.com/ARMmbed/psa-crypto/issues/124).